sources/ElkArte/TemporaryAttachment.php

```diff
@@ -34,30 +33,43 @@ public function __construct($data = null)
 	}
 
 	/**
-	 * Deletes a temporary attachment from the $_SESSION (and the filesystem)
+	 * Deletes a temporary attachment from the filesystem
 	 *
 	 * @param bool $fatal
-	 * @throws \Exception
 	 */
 	public function remove($fatal = true)
 	{
 		$this->data['size'] = 0;
 		$this->data['type'] = '';
 
-		if ($fatal && !file_exists($this->data['tmp_name']))
+		if ($fatal && !$this->fileWritable())
 		{
 			throw new \Exception('attachment_not_found');
 		}
 
-		@unlink($this->data['tmp_name']);
+		try
+		{
+			if (!$this->fileWritable())
+			{
+				throw new \Exception('attachment_not_found');
+			}
+
+			unlink($this->data['tmp_name']);
+		}
+		catch (\Exception $e)
+		{
+			return false;
+		}
+
+		return true;
```

Comment:
Maybe slightly more readable as something like this?

```
if (!$this->fileWritable())
{
	if ($fatal)
	{
		throw new \Exception('attachment_not_found');
	}
	else
	{
		return false;
	}
}

unlink($this->data['tmp_name']);
return true;
```

Unless unlink may throw an exception on failure, not sure.

Reply:
It does :(  ``` An E_WARNING level error will be generated on failure.```  so I was just trying to catch both issues with one catch, either no file or permissions or unlink simply 🤮 for some reason.

Reply:
yay I guess. :laughing: